### DISCUSSION TITLE
Bug with on-demand revalidation? Both the old and new versions of the page are showing

### DISCUSSION THREAD

**Discussion by kristjanmar** [vercel/next.js#34529] [Help] [answered]
I just enabled the new on-demand revalidation feature in NextJS 12.1.0 yesterday.

It looks like calling the API to revalidate pages is working, but sometimes I still see the old version when visiting a page.

As in, I revalidate the page and I navigate to it -- I see the old page. Refreshing the page brings the new version, buf if I navigate away and then back then I get the old one again.

So the only way to produce the new page is to navigate to it and click the refresh button. But navigating to it normally shows the old page.

This also happened when testing in a new incognito window, I believe that rules out any browser caching.

It's as if the edge cache is not being cleared properly after revalidation or something.

Is anyone else seeing this?

**Comment by kristjanmar** [comment]
It seems like when I visit a revalidated page directly, or navigate to it and hit the refresh button, then it shows the correct updated version.

But if I go to the page by navigation -- as in, loading the .json instead of the html file -- then it shows the old version of the page pre-revalidation.

@leerob

**Comment by kristjanmar** [comment, marked as answer]
I see that someone posted an issue already: https://github.com/vercel/next.js/issues/34531

I've uploaded a recording to the issue thread that shows the problem clearly.

**Reply by leerob** [reply]
Going to lock this and let's continue the conversation on the issue, since this is a bug we're fixing 😄
